File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/notify/LocalNotification.kt
package my.ssdid.sdk.domain.notify

import kotlinx.serialization.Serializable

@Serializable
data class LocalNotification(
    val id: String,
    val mailboxId: String,
    val identityName: String?,
    val payload: String,
    val priority: String,
    val receivedAt: String,
    val isRead: Boolean = false
)

File: sdk/ios/SsdidCore/Sources/SsdidCore/Domain/Notify/LocalNotification.swift
import Foundation

public struct LocalNotification: Codable, Identifiable, Equatable {
    public let id: String
    public let mailboxId: String
    public let identityName: String?
    public let payload: String
    public let priority: String
    public let receivedAt: String
    public var isRead: Bool

    public init(
        id: String,
        mailboxId: String,
        identityName: String?,
        payload: String,
        priority: String,
        receivedAt: String,
        isRead: Bool = false
    ) {
        self.id = id
        self.mailboxId = mailboxId
        self.identityName = identityName
        self.payload = payload
        self.priority = priority
        self.receivedAt = receivedAt
        self.isRead = isRead
    }
}
